libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -145,3 +146,51 @@ void ControlFlowAnalyzer::checkUninitializedAccess(CFGNode const* _entry, CFGNod
 		}
 	}
 }
+
+void ControlFlowAnalyzer::checkUnreachable(CFGNode const* _entry, CFGNode const* _exit, CFGNode const* _revert) const
+{
+	std::set<CFGNode const*> reachable;
+	std::set<CFGNode const*> nodesToTraverse;
+
+	// collect all nodes reachable from the entry point
+	nodesToTraverse.insert(_entry);
+	while (!nodesToTraverse.empty())
+	{
+		CFGNode const* node = *nodesToTraverse.begin();
+		nodesToTraverse.erase(nodesToTraverse.begin());
+
+		if (reachable.insert(node).second)
+			for (CFGNode const* exit: node->exits)
+				nodesToTraverse.insert(exit);
```

Comment:
I wonder whether it might be more efficient to already check for `reachable.count(exit)` here.

Reply:
Yes, it probably make sense. I'll consider it when extracting the whole thing.